lib/server.js

```diff
@@ -35,6 +51,22 @@ export class Http3Server extends Http3WebTransport {
     this.stopped = true
   }
 
+  /**
+   * @returns {{ port: number, host: string, family: 'IPv4' | 'IPv6' } | null}
```

Comment:
This implementation works but is very resource intensive. Every call from js to C++ costs a lot of time. So it would be better to keep a copy of the address and port and host on the js side and update it, e.g. when the server goes up.

Reply:
And there is another thing to consider. The objects have members, that should only be access from the libquiche eventloop or the js side on object creation, but never from js side after creation. This applies to any member, that is potentially altered from the libquiche eventlooop. The reason is thread safety without a locking mechanism. And this way of accessing the members break this principle, which may or may not lead to random segfaults.

Reply:
Calling into C++ to get the server address is [what node core does](https://github.com/nodejs/node/blob/main/lib/net.js#L1680-L1692) in the same call?  I take the point that it's slower than caching the values on the js side but this isn't the kind of method you invoke very often.

A interesting point about thread safety, but once the server is running the host and port values aren't going to change?

Reply:
Well, I do not know, for the server probably not. 
But maybe you want the same for the client or other attributes from the server later and these can change,
since quic allows to change IP etc during the connection. So it is better to build in a way, that it can be extended in the future and the philosophy on all part is similar in coding and calling conventions.

Reply:
Another point, we do not know, is if the host and port are always set within the js time. At a later point, this may move into the other thread, and if you call it before the server is up...

Reply:
Seriously, those last seg fault issues were all of the kind, and yes for similar stuff (maybe not this one), tests may help.
And some took days and hours to find.
That is why I am a bit strict in the design pattern of these objects.